src/libraries/System.Private.CoreLib/src/System/Threading/ThreadPoolWorkQueue.cs

```diff
@@ -608,23 +624,68 @@ public void Enqueue(object callback, bool forceGlobal)
             if (_loggingEnabled && FrameworkEventSource.Log.IsEnabled())
                 FrameworkEventSource.Log.ThreadPoolEnqueueWorkObject(callback);
 
-            ThreadPoolWorkQueueThreadLocals? tl;
-            if (!forceGlobal && (tl = ThreadPoolWorkQueueThreadLocals.threadLocals) != null)
+#if CORECLR
+            if (s_prioritizationTest)
             {
-                tl.workStealingQueue.LocalPush(callback);
+                EnqueueForPrioritizationTest(callback, forceGlobal);
             }
             else
+#endif
             {
-                ConcurrentQueue<object> queue =
-                    s_assignableWorkItemQueueCount > 0 && (tl = ThreadPoolWorkQueueThreadLocals.threadLocals) != null
-                        ? tl.assignedGlobalWorkItemQueue
-                        : workItems;
-                queue.Enqueue(callback);
+                ThreadPoolWorkQueueThreadLocals? tl;
+                if (!forceGlobal && (tl = ThreadPoolWorkQueueThreadLocals.threadLocals) != null)
+                {
+                    tl.workStealingQueue.LocalPush(callback);
+                }
+                else
+                {
+                    ConcurrentQueue<object> queue =
+                        s_assignableWorkItemQueueCount > 0 && (tl = ThreadPoolWorkQueueThreadLocals.threadLocals) != null
+                            ? tl.assignedGlobalWorkItemQueue
+                            : workItems;
+                    queue.Enqueue(callback);
+                }
             }
 
             EnsureThreadRequested();
         }
 
+#if CORECLR
+        [MethodImpl(MethodImplOptions.NoInlining)]
```

Comment:
Why the no inlining? I was actually expecting to see aggressive inlining, since it's only used in one place and when s_prioritizationTest is true the else block above will be JIT compiled away.

Reply:
I was trying to avoid bloating the method's code size (R2R code size, jitted code size when tiering is disabled) by separating out the uncommon path. I've seen the method get inlined before, so didn't want to affect that much either.

Reply:
Ok